external_chi2/optimal_observables/interface_oos.py

```diff
@@ -27,13 +27,14 @@ def __init__(self, coefficients):
             '{collider}_ww_semilep_365': 'invcov_{collider}_ww_semilep_365.dat'
         }
 
+        self.invcovs = [np.loadtxt(current_file_path / path.format(collider=collider)) for path in self.datasets.values()]
+
         self.n_dat = len(oo_wc_basis)
 
     def compute_chi2(self, coefficient_values):
 
         chi2_value = 0
-        for invcov_path in self.datasets.values():
-            invcov = np.loadtxt(current_file_path / invcov_path.format(collider=collider))
+        for invcov in self.invcovs:
             chi2_value += np.linalg.multi_dot(
                 [coefficient_values, self.project.T, invcov, self.project, coefficient_values])
```

Comment:
why don't we optimise further her, the self.project.T @ invcov @ self.project can be computed ones for all.

Also, let's use jnp.linalg?